packages/block-library/src/image/interactivity.js

```diff
@@ -142,6 +158,30 @@ store( {
 	effects: {
 		core: {
 			image: {
+				setCurrentSrc: ( { context, ref } ) => {
+					if ( ref.complete ) {
+						context.core.image.imageLoaded = true;
+						context.core.image.imageCurrentSrc = ref.currentSrc;
+					} else {
+						ref.addEventListener( 'load', function () {
+							context.core.image.imageLoaded = true;
+							context.core.image.imageCurrentSrc =
+								this.currentSrc;
+						} );
+					}
+				},
+				preloadLightboxImage: ( { context, ref } ) => {
+					ref.addEventListener( 'mouseover', () => {
```

Comment:
Hey @artemiomorales, out of curiosity, why did you use `useEffect` instead of `data-wp-on--mouseover`?

Reply:
Thanks for catching this! It's an oversight on my part.

Reply:
@luisherranz Update: I decided to break this out into a [separate PR](https://github.com/WordPress/gutenberg/pull/52067) for easier review and approval.